lib/Frontend/ModuleInterfaceBuilder.cpp

```diff
@@ -191,13 +191,6 @@ bool ModuleInterfaceBuilder::buildSwiftModuleInternal(
     InputInfo.getPrimarySpecificPaths().SupplementaryOutputs;
     StringRef OutPath = OutputInfo.ModuleOutputPath;
 
-    // Bail out if we're going to use the standard library but can't load it. If
-    // we don't do this before we try to build the interface, we could end up
-    // trying to rebuild a broken standard library dozens of times due to
-    // multiple calls to `ASTContext::getStdlibModule()`.
-    if (SubInstance.loadStdlibIfNeeded())
-      return std::make_error_code(std::errc::not_supported);
-
```

Comment:
@nkcsgexi @artemcm Mind double checking that it's okay this check is now done for all clients of `InterfaceSubContextDelegateImpl::runInSubCompilerInstance`? In particular, it *appears* that dependency scanning uses it, and `ModuleInterfaceLoaderOptions` seems to be currently configured with actions that would require a stdlib to be loaded, so if I'm reading it right, this may now cause the stdlib to be loaded when scanning dependencies for a swiftinterface. Is that correct?

Reply:
Actually, I believe it’s the other way round. Previously, we were always loading the stdlib for dependency scanning https://github.com/apple/swift/blob/0bd5ddb73acb88d72ff04ac88cb36e34dfd04b05/test/ScanDependencies/module_deps_cache_reuse.swift#L51
but dependency scanning isn’t listed as requiring a stdlib to be loaded https://github.com/apple/swift/blob/0bd5ddb73acb88d72ff04ac88cb36e34dfd04b05/lib/Frontend/FrontendOptions.cpp#L99
so it might not be loaded anymore after this change. Should we change `ScanDependencies` to return `true` in `doesActionRequireSwiftStandardLibrary` and thus also make it return `false` for `shouldActionOnlyParse`?

Reply:
Actually, I believe it’s the other way round. Previously, we were always loading the stdlib for dependency scanning https://github.com/apple/swift/blob/0bd5ddb73acb88d72ff04ac88cb36e34dfd04b05/test/ScanDependencies/module_deps_cache_reuse.swift#L51
but dependency scanning isn’t listed as requiring a stdlib to be loaded https://github.com/apple/swift/blob/0bd5ddb73acb88d72ff04ac88cb36e34dfd04b05/lib/Frontend/FrontendOptions.cpp#L99
so it might not be loaded anymore after this change. Should we change `ScanDependencies` to return `true` in `doesActionRequireSwiftStandardLibrary` and thus also make it return `false` for `shouldActionOnlyParse`?

Reply:
@ahoppen As I understand it, dependency scanning shouldn't need to load modules, it should still be able to pick up the implicit stdlib import though. I actually realize now this means that `doesActionRequireSwiftStandardLibrary` *shouldn't* be sunk into `getImplicitStdlibKind`, as when scanning dependencies, we need to know whether an implicit stdlib import is available, but we shouldn't actually load it.

So I believe `doesActionRequireSwiftStandardLibrary` & `shouldActionOnlyParse` are correct, but the logic that kicks of dependency scanning needs to be updated to not use the `Typecheck` or `EmitModuleOnly` frontend actions via `ModuleInterfaceLoaderOptions`, e.g:

https://github.com/apple/swift/blob/b02dc65e911e7aa4aff519ebaef7f092fedb86dd/lib/DependencyScan/ScanDependencies.cpp#L1381

Or `runInSubCompilerInstance` needs to otherwise be changed to make sure not to load the stdlib for dep scanning.